server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -565,6 +576,30 @@ private static Map<String, Mapper> buildMergedMappers(
             }
             return Map.copyOf(mergedMappers);
         }
+
+        private static void addNewObjectMapper(
```

Comment:
Thanks, this method and extracting #104578 are great ideas and do help following what is happening in this PR.